Add key-value-db and sqlite-db features, separate wallet directories

notmandatory opened the pull request:
### Description

- Add distinct `key-value-db` and `sqlite-db` features, keep default as `key-value-db`

- Put cached wallet data in separate directories: `~/.bdk-bitcoin/<wallet_name>`
- Put compact filter data in `<wallet_name>/compact_filters`
- Depending on the db used put cached wallet data in: `<wallet_name>/wallet.sled/` or `<wallet_name>/wallet.sqlite`

### Notes to the reviewers

This change will help test BDK with different databases, in particular for manually testing DB migrations such as in https://github.com/bitcoindevkit/bdk/pull/502.

### Checklists

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk-cli/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

#### New Features:

* [ ] I've added tests for the new feature
* [x] I've added docs for the new feature
* [x] I've updated `CHANGELOG.md`

notmandatory commented:
@rajarshimaitra can you take a look at this one?

rajarshimaitra commented on src/bdk_cli.rs:
```diff
@@ -112,25 +115,75 @@ fn prepare_home_dir() -> Result<PathBuf, Error> {
+
+/// Prepare blockchain data directory (for compact filters)
+#[cfg(feature = "compact_filters")]
+fn prepare_bc_dir(wallet_name: &str) -> Result<PathBuf, Error> {
```
Just a question: something like this could also be used for auto regtest node datadir right?

notmandatory replied:
Yes! I'd like to put the auto regtest data someplace like:
* `~/.bdk-bitcoin/<wallet_name>/regtest/bitcoind/` and
* `~/.bdk-bitcoin/<wallet_name>/regest/electrsd/`.

rajarshimaitra reviewed (commented):
Review ACK https://github.com/bitcoindevkit/bdk-cli/pull/71/commits/2e3801151afeb81767605b674c643e8b0a42b454

Looks good for now.. Will review again when additive features stuff is done..

notmandatory commented:
@rajarshimaitra I changed my mind again on the additive features thing. I still think it'd be good to try but will big a biggish change (if it's even possible) for bdk-cli and probably not worth the effort right now.

This PR is now rebased and should be ready to go.

rajarshimaitra commented on src/bdk_cli.rs:
```diff
@@ -100,25 +103,75 @@ fn prepare_home_dir() -> Result<PathBuf, Error> {
+    #[cfg(feature = "sqlite-db")]
+    db_dir.push("wallet.sqlite");
+
+    #[cfg(not(feature = "sqlite-db"))]
+    if !db_dir.exists() {
+        info!("Creating database directory {}", db_dir.as_path().display());
+        fs::create_dir(&db_dir).map_err(|e| Error::Generic(e.to_string()))?;
+    }
```
I did not understand the reason of feature guard here. For both sled and sqlite we would want to check if the dir exists right??

notmandatory replied:
The reason is that sqlite creates only a single file and doesn't need a directory to exist to put it's files in.

rajarshimaitra reviewed (commented):
Ack 00454be42be22907d4d9c959483e79dc662de71c

With just one question..
